Deployment steps for usb-camera-demo need slight edit in order to work on an Azure VM due to missing kernel modules

Installing the [usb-camera-demo sample](https://docs.akri.sh/demos/usb-camera-demo) on an Azure VM (I used Ubuntu 18.04 LTS) doesn't work with the instructions as they are now. An error occurs when trying to install the v4l2loopback kernel module due to the VM missing certain required kernel modules (e.g. usb device drivers). Here are the steps I did:

az vm create \
  --resource-group ${RESOURCE_GROUP} \
  --name ${VM_NAME} \
  --image UbuntuLTS \
  --admin-username ${VM_USER_NAME} \
  --admin-password ${VM_PASSWORD} \
  --public-ip-sku Standard

and in step 2 under “Setting up mock udev video devices” got this error:

...
sudo dpkg -i v4l2loopback-dkms_0.12.5-1_all.deb
Selecting previously unselected package v4l2loopback-dkms.
(Reading database ... 82547 files and directories currently installed.)
Preparing to unpack v4l2loopback-dkms_0.12.5-1_all.deb ...
Unpacking v4l2loopback-dkms (0.12.5-1) ...
Setting up v4l2loopback-dkms (0.12.5-1) ...
Loading new v4l2loopback-0.12.5 DKMS files...
Building for 5.4.0-1072-azure
Building initial module for 5.4.0-1072-azure
Error! Bad return status for module build on kernel: 5.4.0-1072-azure (x86_64)
Consult /var/lib/dkms/v4l2loopback/0.12.5/build/make.log for more information.

The log shows:

DKMS make.log for v4l2loopback-0.12.5 for kernel 5.4.0-1072-azure (x86_64)
Thu Mar 17 18:19:28 UTC 2022
Building v4l2-loopback driver...
make -C /lib/modules/5.4.0-1072-azure/build M=/var/lib/dkms/v4l2loopback/0.12.5/build modules
make[1]: Entering directory '/usr/src/linux-headers-5.4.0-1072-azure'
  CC [M]  /var/lib/dkms/v4l2loopback/0.12.5/build/v4l2loopback.o
  Building modules, stage 2.
  MODPOST 1 modules
ERROR: "video_ioctl2" [/var/lib/dkms/v4l2loopback/0.12.5/build/v4l2loopback.ko] undefined!
ERROR: "__video_register_device" [/var/lib/dkms/v4l2loopback/0.12.5/build/v4l2loopback.ko] undefined!
ERROR: "v4l2_ctrl_new_custom" [/var/lib/dkms/v4l2loopback/0.12.5/build/v4l2loopback.ko] undefined!
ERROR: "v4l2_ctrl_handler_init_class" [/var/lib/dkms/v4l2loopback/0.12.5/build/v4l2loopback.ko] undefined!
ERROR: "video_device_release" [/var/lib/dkms/v4l2loopback/0.12.5/build/v4l2loopback.ko] undefined!
ERROR: "video_device_alloc" [/var/lib/dkms/v4l2loopback/0.12.5/build/v4l2loopback.ko] undefined!
ERROR: "v4l2_device_register" [/var/lib/dkms/v4l2loopback/0.12.5/build/v4l2loopback.ko] undefined!
ERROR: "v4l2_ctrl_handler_free" [/var/lib/dkms/v4l2loopback/0.12.5/build/v4l2loopback.ko] undefined!
ERROR: "v4l2_device_unregister" [/var/lib/dkms/v4l2loopback/0.12.5/build/v4l2loopback.ko] undefined!
ERROR: "video_unregister_device" [/var/lib/dkms/v4l2loopback/0.12.5/build/v4l2loopback.ko] undefined!
ERROR: "video_devdata" [/var/lib/dkms/v4l2loopback/0.12.5/build/v4l2loopback.ko] undefined!
scripts/Makefile.modpost:93: recipe for target '__modpost' failed
make[2]: *** [__modpost] Error 1
Makefile:1675: recipe for target 'modules' failed
make[1]: *** [modules] Error 2
make[1]: Leaving directory '/usr/src/linux-headers-5.4.0-1072-azure'
Makefile:42: recipe for target 'v4l2loopback.ko' failed
make: *** [v4l2loopback.ko] Error 2

I got it to work (thanks [kate-goldenring](https://github.com/kate-goldenring) for the help) by adjusting the deployment steps slightly as follows:

sudo apt update
sudo apt -y install linux-modules-extra-azure
sudo apt -y install linux-headers-$(uname -r)
sudo apt -y install linux-modules-extra-$(uname -r)
sudo reboot now
sudo apt -y install dkms
curl http://deb.debian.org/debian/pool/main/v/v4l2loopback/v4l2loopback-dkms_0.12.5-1_all.deb -o v4l2loopback-dkms_0.12.5-1_all.deb
sudo dpkg -i v4l2loopback-dkms_0.12.5-1_all.deb

It might be best to add to the demo that it assumes the VM being used supports the proper kernel modules, and perhaps incorporate the above working steps in some section specific to an Azure VM.

--- a/docs/demos/usb-camera-demo.md
+++ b/docs/demos/usb-camera-demo.md
@@ -14,15 +14,19 @@ The following will be covered in this demo:
 
 ## Setting up mock udev video devices
 
-1. Acquire an Ubuntu 20.04 LTS, 18.04 LTS or 16.04 LTS environment to run the
+1. Acquire an Ubuntu 20.04 LTS, 18.04 LTS or 16.04 LTS environment to run the commands. This demo assumes that the VM
+   being used supports the proper kernel modules, which may not be the case if using a cloud-based VM which sometimes
+   have been slimmed down to remove unnecessary modules such as for USB devices. For example, on an Ubuntu 20.04 VM in
+   Azure, the following prerequisite step is needed to add the necessary kernel modules:
 
-   commands. If you would like to deploy the demo to a cloud-based VM, see the
+   ```sh
+   sudo apt update
+   sudo apt -y install linux-modules-extra-azure
+   ```
 
-   guides on Akri's HackMD for [DigitalOcean](https://hackmd.io/@akri/Hyz1GW1gY) or [Google Compute
-
-   Engine](https://hackmd.io/@akri/rJHdQWJeF) (and you can skip the rest of the steps in
-
-   this document). Note, these guides are unmaintained and may not be up to date.
+   > Note: There are also guides Akri's HackMD for running the demo on [DigitalOcean](https://hackmd.io/@akri/Hyz1GW1gY)
+   > and [Google Compute Engine](https://hackmd.io/@akri/rJHdQWJeF) (and you can skip the rest of the steps in this
+   > document). Note, these guides are unmaintained and may not be up to date.
 
 1. To setup fake usb video devices, install the v4l2loopback kernel module and its prerequisites. Learn more about v4l2 loopback [here](https://github.com/umlaeute/v4l2loopback)
 

